sway/config.c

```diff
@@ -574,7 +574,13 @@ bool load_include_configs(const char *path, struct sway_config *config,
 	char **w = p.we_wordv;
 	size_t i;
 	for (i = 0; i < p.we_wordc; ++i) {
-		load_include_config(w[i], parent_dir, config, swaynag);
+		bool found = load_include_config(w[i], parent_dir, config, swaynag);
+		if (!found) {
+			wordfree(&p);
+			free(parent_path);
+			free(wd);
+			return false;
```

Comment:
I think we also need to restore the CWD here. An approach like this would probably work:

```c
bool success = true;

for (…) {
	success = load_include_config(…);
	if (!success) {
		goto out;
	}
}

out:
free(…);
chdir(…); // TODO: error handling

return success;
```

Reply:
What about the early `return false` on line 571? The CWD doesn't appear to be restored there.

Reply:
Ah, right. Could you fix this one as well?